src/libstore/unix/build/freebsd-derivation-builder.cc

```diff
@@ -0,0 +1,501 @@
+                throw SysError("stat");
+            }
+
+            // mount points must exist and be the right type
+            if (S_ISDIR(stat_buf.st_mode)) {
+                createDirs(path);
+            } else {
+                createDirs(dirOf(path));
+                writeFile(path, "");
+            }
+
+            struct iovec iov[8] = {
+                iovFromStaticSizedString("fstype"),
+                iovFromStaticSizedString("nullfs"),
+                iovFromStaticSizedString("fspath"),
+                iovFromDynamicSizeString(path),
+                iovFromStaticSizedString("target"),
+                iovFromDynamicSizeString(i.second.source),
+                iovFromStaticSizedString("errmsg"),
+                iovFromStaticSizedString(errmsg),
+            };
+            if (nmount(iov, 8, 0) < 0) {
+                throw SysError("Failed to mount nullfs for %1% - %2%", path, errmsg);
+            }
+            autoDelMounts.push_back(std::make_shared<AutoUnmount>(path));
+        }
+
+        /* Fixed-output derivations typically need to access the
+           network, so give them access to /etc/resolv.conf and so
+           on. */
+        if (!derivationType.isSandboxed()) {
+            // Only use nss functions to resolve hosts and
+            // services. Don’t use it for anything else that may
+            // be configured for this system. This limits the
+            // potential impurities introduced in fixed-outputs.
+            writeFile(chrootRootDir + "/etc/nsswitch.conf", "hosts: files dns\nservices: files\n");
+
+            /* N.B. it is realistic that these paths might not exist. It
+               happens when testing Nix building fixed-output derivations
+               within a pure derivation. */
+            for (std::filesystem::path path : {"/etc/resolv.conf", "/etc/services", "/etc/hosts"}) {
+                if (pathExists(path)) {
+                    // This means if your network config changes during a FOD build,
+                    // the DNS in the sandbox will be wrong. However, this is pretty unlikely
+                    // to actually be a problem, because FODs are generally pretty fast,
+                    // and machines with often-changing network configurations probably
+                    // want to run resolved or some other local resolver anyway.
+                    //
+                    // There's also just no simple way to do this correctly, you have to manually
+                    // inotify watch the files for changes on the outside and update the sandbox
+                    // while the build is running (or at least that's what Flatpak does).
+                    //
+                    // I also just generally feel icky about modifying sandbox state under a build,
+                    // even though it really shouldn't be a big deal. -K900
+                    copyFile(path, std::filesystem::path{chrootRootDir} / path.relative_path(), false, true);
```

Comment:
This way it's slightly more readable.

```suggestion
                    copyFile(path, std::filesystem::path{chrootRootDir} / path.relative_path(), /*andDelete=*/false, /*contents=*/true);
```